### PR TITLE
fix(gatsby-source-filesystem): Use new FileType methods to read file extension

### LINKED ISSUE #21901: createRemoteFileNode crashes when url doesn't have an extension
## Description

calling `createRemoteFileNode` with an url (<https://fakeimg.pl/300x300/?text=2>) without an extension crashes.

```
Unexpected error value: "failed to process https://fakeimg.pl/300x300/?text=2\nError: ENOENT: no such file or directory, stat '.../.cache/gatsby-source-filesystem/7a1221ea4c61551f928908dd873a3738/300x300'"
```
I think it has something to do with how the caching is handled.

### Steps to reproduce

1. use `createRemoteFileNode` to create remote file nodes from an url without an ext
2. use the remote file node in eg. sharp-image
3. the first build goes through
4. all builds afterwards crash

### Environment

```
  System:
    OS: Linux 5.3 Ubuntu 18.04.3 LTS (Bionic Beaver)
    CPU: (8) x64 AMD Ryzen 5 2500U with Radeon Vega Mobile Gfx
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 13.9.0 - /usr/bin/node
    Yarn: 1.22.0 - ~/bin/yarn
    npm: 6.13.7 - /usr/bin/npm
  Languages:
    Python: 2.7.15+ - /usr/bin/python
  Browsers:
    Firefox: 73.0.1
  npmPackages:
    gatsby: ^2.19.21 => 2.19.24
    gatsby-cli: ^2.9.0 => 2.9.0
    gatsby-image: ^2.2.41 => 2.2.42
    gatsby-plugin-graphql-codegen: ^2.2.2 => 2.2.3
    gatsby-plugin-react-helmet: ^3.1.21 => 3.1.22
    gatsby-plugin-react-svg: ^3.0.0 => 3.0.0
    gatsby-plugin-sass: ^2.1.27 => 2.1.29
    gatsby-plugin-sharp: ^2.4.5 => 2.4.5
    gatsby-source-filesystem: ^2.1.48 => 2.1.48
    gatsby-transformer-json: ^2.2.26 => 2.2.26
    gatsby-transformer-sharp: ^2.3.14 => 2.3.16
```
### Repro
<https://github.com/chpio/gatsby-remote-wo-ext> commit https://github.com/chpio/gatsby-remote-wo-ext/commit/8d898f6915221495e092798e5bfcbacdd3faaf30

### PR DESCRIPTION
**Pull request by ynnoj**
## Description

Uses [`fromFile()`](https://github.com/sindresorhus/file-type#filetypefromfilefilepath), [`fromBuffer()`](https://github.com/sindresorhus/file-type#filetypefrombufferbuffer) methods to infer filetype/extension from the file/buffer, if not specified.

This fixes an issue where `createRemoteFileNode` may fail on URLs without an extension in the path.

### Documentation

* N/A

## Related Issues

* Fixes #21901

### CODE REVIEW

**Review by pvdz — APPROVED** [review]
lgtm. Thank you :purple_heart:

File: packages/gatsby-source-filesystem/package.json [hunk 1]
```diff
@@ -10,15 +10,14 @@
     "@babel/runtime": "^7.11.2",
     "better-queue": "^3.8.10",
     "chokidar": "^3.4.2",
-    "file-type": "^12.4.2",
+    "file-type": "^16.0.0",
```
Comment (pvdz) [inline packages/gatsby-source-filesystem/package.json:13]: You need to go and find all other references to `fileType` in this package as the API changed (see tests). Run `yarn test` to confirm. In particular in gatsby-source-filesystem/src/create-file-node-from-buffer.js and gatsby-source-filesystem/src/create-remote-file-node.js

**Review by pvdz — CHANGES_REQUESTED** [review]
Actually, tests show that bumping the package requires more changes

File: packages/gatsby-source-filesystem/package.json (hunk 1 above)
Comment (ynnoj) [reply packages/gatsby-source-filesystem/package.json:13]: @pvdz Thanks, will fix!

**Comment by ynnoj** [comment]
@pvdz Test suite still failing in CI, but seems unrelated to any changes in this PR. 

FWIW, test suite ran and passed locally.

**Comment by Rikpat** [comment]
What is the state of this PR? I'm trying to use createRemoteFileNode to get profile images from Azure DevOps and the format is like aad.[somecode] and that doesn't work with the current createRemoteFileNode

**Comment by ynnoj** [comment]
@pvdz @wardpeet Anything I can do to help move this along?

**Comment by pvdz** [comment]
Let me rebase this for you and see whether tests are broken or just need to be rebased

**Comment by pvdz** [comment]
--> https://github.com/gatsbyjs/gatsby/pull/28156

**Pull request closed without merging**